Userland/Libraries/LibSemVer/SemVer.cpp

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright (c) 2023, Gurkirat Singh <tbhaxor@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/GenericLexer.h>
+#include <AK/StringBuilder.h>
+
+#include <LibRegex/RegexMatcher.h>
+#include <LibRegex/RegexParser.h>
+#include <LibSemVer/Semver.h>
+
+namespace SemVer {
+String SemVer::suffix() const
+{
+    StringBuilder sb;
+    if (m_prerelease.bytes().size() > 0) {
+        sb.append('-');
+        sb.append(m_prerelease);
+    }
+    if (m_buildmetadata.bytes().size() > 0) {
+        sb.append('+');
+        sb.append(m_buildmetadata);
+    }
+    return MUST(sb.to_string());
+}
+
+String SemVer::to_string() const
+{
+    return MUST(String::formatted("{}{}{}{}{}{}", m_major, m_mmp_sep, m_minor, m_mmp_sep, m_patch, suffix()));
+}
+
+ReleaseType SemVer::release_type() const
+{
+    auto prerel_identifiers = MUST(MUST(m_prerelease.to_lowercase()).split('.'));
+    auto is_alpha = prerel_identifiers.contains_slow("alpha"_string);
+    auto is_beta = prerel_identifiers.contains_slow("beta"_string);
+
+    if (is_alpha && is_beta) {
+        return ReleaseType::AlphaBeta;
+    }
+
+    if (is_alpha) {
+        return ReleaseType::Alpha;
+    }
+
+    if (is_beta) {
+        return ReleaseType::Beta;
+    }
+
+    if (prerel_identifiers.contains_slow("prerelease"_string)) {
+        return ReleaseType::PreRelease;
+    }
+
+    // strict release candidate check
+    if (prerel_identifiers.contains_slow("rc"_string)) {
+        return ReleaseType::ReleaseCandidate;
+    }
+
+    return ReleaseType::Stable;
+}
```

Comment:
This looks like it's not actually standardized in any way (`alpha` and `beta` aren't special, anything that has a `prerel_identifier` is considered a prerelease already, etc.), so I'd recommend not introducing such a function (for now) and to just rely on correct implementation of `satisfies`.